File: v3-sandbox/prototype-keys-java/src/main/java/org/hiero/keys/io/ByteImportEncoding.java
package org.hiero.keys.io;

/**
 * Import encodings for string representations of bytes.
 */
public enum ByteImportEncoding {
    HEX,
    BASE64;

    public byte[] decode(final String value) {
        return switch (this) {
            case HEX -> org.hiero.keys.impl.Hex.decode(value);
            case BASE64 -> java.util.Base64.getDecoder().decode(value);
        };
    }
}

File: v3-sandbox/prototype-python/src/hiero_sdk_v3/keys/io/byte_import_encoding.py
import base64
from enum import Enum


class ByteImportEncoding(Enum):
    HEX = "HEX"
    BASE64 = "BASE64"

    def decode(self, value: str) -> bytes:
        if self == ByteImportEncoding.HEX:
            return bytes.fromhex(value)
        return base64.b64decode(value)
